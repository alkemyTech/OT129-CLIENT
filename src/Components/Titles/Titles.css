.banner {
  position: relative;
  text-align: center;
}

.image-banner {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: center;
}

.text-banner {
  z-index: 1;
}

.title-banner {
  font-weight: 600;
  font-size: 2.3rem;
  text-transform: uppercase;
}

