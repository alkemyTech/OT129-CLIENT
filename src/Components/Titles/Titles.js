import React from "react";

const Titles = () => {
  return <></>;
};

export default Titles;
