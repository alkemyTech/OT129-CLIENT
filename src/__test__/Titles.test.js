import React from "react";
import { render } from "@testing-library/react";

import Titles from "../Components/Titles/Titles";

const defaultProps = {
  title: "Titulo",
  description: "Descripcion",
};

describe("<Titles />", () => {
  it("Should render without crash", () => {
    const { asFragment } = render(<Titles {...defaultProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("Should render  title and description", () => {
    const { getByText } = render(<Titles {...defaultProps} />);

    expect(getByText(defaultProps.title)).toBeInTheDocument();
    expect(getByText(defaultProps.description)).toBeInTheDocument();
  });

  it("Should render image by default", () => {
    const { getByAltText } = render(<Titles {...defaultProps} />);

    const imageComponent = getByAltText(defaultProps.title);

    expect(imageComponent.src).toContain("background-title.jpg");
  });

  it("Should render image by props", () => {
    const image = "imagePrueba";
    const { getByAltText } = render(<Titles {...defaultProps} image={image} />);

    const imageComponent = getByAltText(defaultProps.title);

    expect(imageComponent.src).toContain(image);
  });
});
