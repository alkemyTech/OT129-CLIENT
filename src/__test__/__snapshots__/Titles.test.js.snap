// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Titles /> Should render without crash 1`] = `
<DocumentFragment>
  <div
    class="container-fluid d-flex justify-content-center align-items-center banner py-5"
  >
    <img
      alt="Titulo"
      class="image-banner"
      src="background-title.jpg"
    />
    <div
      class="row"
    >
      <div
        class="container text-banner"
      >
        <div
          class="row row-banner"
        >
          <div
            class="col-12"
          >
            <h1
              class="mb-0 title-banner"
            >
              Titulo
            </h1>
            <p
              class="mb-0 mt-3"
            >
              Descripcion
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
